aes-kw/src/kek.rs

```diff
@@ -0,0 +1,225 @@
+use aes::cipher::{BlockDecrypt, BlockEncrypt, NewBlockCipher};
+use aes::BlockCipher;
+use generic_array::typenum::{U16, U24, U32};
+use generic_array::GenericArray;
+
+const IV_LEN: usize = 8;
+const IV: [u8; IV_LEN] = [0xA6, 0xA6, 0xA6, 0xA6, 0xA6, 0xA6, 0xA6, 0xA6];
+
+use crate::Error;
+
+/// A KEK that can be used to wrap and unwrap.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum Kek {
+    /// 128bit sized Kek
+    Aes128(GenericArray<u8, U16>),
+    /// 192bit sized Kek
+    Aes192(GenericArray<u8, U24>),
+    /// 256bit sized Kek
+    Aes256(GenericArray<u8, U32>),
```

Comment:
An alternative to this sort of explicit enumeration would be to make the inner type generic. As an example, here's how the `aes-gcm` crate does it:

https://github.com/RustCrypto/AEADs/blob/master/aes-gcm/src/lib.rs#L176

This does have the disadvantage that it makes it possible to use a non-AES block cipher, but also the advantage that the user can supply an alternative AES implementation, for example an embedded hardware accelerator/coprocessor.

If you store the initialized instance, it will also avoid repeating any sort of round key setup which would provide better performance.

Reply:
great idea, this makes the code much cleaner